src/test/kotlin/org/dif/crypto/key/SenderKeySelectorTest.kt

```diff
@@ -0,0 +1,289 @@
+        val senderKeySelector = SenderKeySelector(DIDDocResolverMock(), AliceSecretResolverMock())
+        val (from, to) = senderKeySelector.findAuthCryptKeys(JWM.ALICE_DID, "did:example:bob#key-x25519-3")
+
+        val expected = Pair(
+            "did:example:alice#key-x25519-1",
+            listOf(
+                "did:example:bob#key-x25519-3"
+            )
+        )
+
+        assertEquals(expected.first, from.id)
+        assertContentEquals(expected.second, to.map { it.id })
+    }
+
+    @Test
+    fun `Test find auth crypto keys by DID URL`() {
+        val senderKeySelector = SenderKeySelector(DIDDocResolverMock(), AliceSecretResolverMock())
+        val (from, to) = senderKeySelector.findAuthCryptKeys("did:example:alice#key-x25519-1", "did:example:bob#key-x25519-3")
+
+        val expected = Pair(
+            "did:example:alice#key-x25519-1",
+            listOf(
+                "did:example:bob#key-x25519-3"
+            )
+        )
+
+        assertEquals(expected.first, from.id)
+        assertContentEquals(expected.second, to.map { it.id })
+    }
+
+    @Test
+    fun `Test signing key not found by DID`() {
+        val senderKeySelector = SenderKeySelector(DIDDocResolverMock(), AliceSecretResolverMock())
+
+        val actual = assertFailsWith<DIDDocException> {
+            senderKeySelector.findSigningKey(JWM.BOB_DID)
+        }
+
+        assertEquals("The DID Doc '${JWM.BOB_DID}' does not contain compatible 'authentication' verification methods", actual.message)
+    }
+
+    @Test
+    fun `Test key not found by DID URL`() {
+        val senderKeySelector = SenderKeySelector(DIDDocResolverMock(), AliceSecretResolverMock())
+        val expected = "The Secret 'did:example:alice#key-x25519-3' not found"
+        val didUrl = "did:example:alice#key-x25519-3"
+
+        run {
+            val actual = assertFailsWith<SecretNotFoundException> {
+                senderKeySelector.findSigningKey(didUrl)
+            }
+
+            assertEquals(expected, actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<SecretNotFoundException> {
+                senderKeySelector.findAuthCryptKeys(didUrl, "did:example:bob#key-x25519-1")
+            }
+
+            assertEquals(expected, actual.message)
+        }
+
+        run {
+            val actual = assertFailsWith<SecretNotFoundException> {
+                senderKeySelector.findAuthCryptKeys(didUrl, JWM.BOB_DID)
+            }
+
+            assertEquals(expected, actual.message)
+        }
+    }
+
+    @Test
+    fun `Test verification method not found by DID URL`() {
+        val senderKeySelector = SenderKeySelector(DIDDocResolverMock(), AliceSecretResolverMock())
+        val expected = "Verification method 'did:example:bob#key-4' not found in DID Doc 'did:example:bob'"
+        val didUrl = "did:example:bob#key-4"
+
+        run {
+            val actual = assertFailsWith<DIDDocException> {
```

Comment:
I would expect DIDUrlNotFoundException here.

Reply:
I think we need conversation here. The reason why I throw DIDDocException it's because did doc was found, but verification method not. IMHO it's a DID Doc problem, not URL.